app/api/userdata/route.js

```diff
@@ -45,4 +45,34 @@ export async function POST(req) {
     } catch(error) {
         return NextResponse.json( error )
     }
+}
+
+export async function PATCH(req) {
+    const newData = await req.json();
+    try {
+        if (newData.field==='First name') {
```

Comment:
are you only able to update one field at a time? what happens if you want to update first name and last name at the same time? does that functionality exist? 

If you try to update both at the same time with this current function, it'll hit the first if statement and not proceed to the second statement to update the last name.